systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/ManualRollingUpdateST.java

```diff
@@ -0,0 +1,252 @@
+ */
+package io.strimzi.systemtest.rollingupdate;
+
+import io.fabric8.kubernetes.api.model.Event;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.api.kafka.model.KafkaUser;
+import io.strimzi.operator.common.Annotations;
+import io.strimzi.systemtest.BaseST;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.kafkaclients.internalClients.InternalKafkaClient;
+import io.strimzi.systemtest.resources.KubernetesResource;
+import io.strimzi.systemtest.resources.ResourceManager;
+import io.strimzi.systemtest.resources.crd.KafkaClientsResource;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.systemtest.resources.crd.KafkaTopicResource;
+import io.strimzi.systemtest.resources.crd.KafkaUserResource;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaTopicUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaUserUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.StatefulSetUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import io.strimzi.test.TestUtils;
+import io.strimzi.test.timemeasuring.Operation;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import static io.strimzi.systemtest.Constants.INTERNAL_CLIENTS_USED;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.systemtest.k8s.Events.Created;
+import static io.strimzi.systemtest.k8s.Events.Pulled;
+import static io.strimzi.systemtest.k8s.Events.Scheduled;
+import static io.strimzi.systemtest.k8s.Events.Started;
+import static io.strimzi.systemtest.matchers.Matchers.hasAllOfReasons;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsMapContaining.hasEntry;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Tag(REGRESSION)
+@Tag(INTERNAL_CLIENTS_USED)
+class ManualRollingUpdateST extends BaseST {
+
+    private static final Logger LOGGER = LogManager.getLogger(RollingUpdateST.class);
+
+    static final String NAMESPACE = "manual-rolling-update-cluster-test";
+
+    /**
+     * This test cover case, when KafkaRoller will not roll Kafka pods, because created topic doesn't meet requirements for roll remaining pods
+     * 1. Deploy kafka cluster with 4 pods
+     * 2. Create topic with 4 replicas
+     * 3. Scale down kafka cluster to 3 replicas
+     * 4. Trigger rolling update for Kafka cluster
+     * 5. Rolling update will not be performed, because topic which we created had some replicas on deleted pods - manual fix is needed in that case
+     */
+    @Test
+    void testKafkaWontRollUpBecauseTopic() {
```

Comment:
I guess this will conflict with #3041?

Reply:
Yes I'm waiting for merge, then I'll resolve conflicts

Reply:
Yes. I am sure Lukas will deal with it :D